pkg/trustedresources/verify.go

```diff
@@ -186,15 +200,21 @@ func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 		// if this policy fails the verification and the mode is not "warn", should return error directly. No need to check other policies
 		if !passVerification {
 			if p.Spec.Mode == v1alpha1.ModeWarn {
-				logger := logging.FromContext(ctx)
+				failWarnPolicyVerification = true
 				logger.Warnf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
 			} else {
 				// if the mode is "enforce" or not set, return error.
-				return fmt.Errorf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
+				return &VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())}
 			}
 		}
 	}
-	return nil
+	if getKeysErrorFromWarnPolicy {
```

Comment:
If there are 2 warn mode policies, one doesn't have keys, the other has invalid keys. The warning from no keys policy (warningGetKeysFromWarnPolicy) will be returned due to the code execution order. 

I think this is a complicated and rare use case, maybe we could focus on high priorities then figure out those cases.